src/createRestyleFunction.ts

```diff
@@ -1,5 +1,26 @@
-import {BaseTheme, RestyleFunction, RNStyleProperty} from './types';
-import {getResponsiveValue, StyleTransformFunction} from './responsiveHelpers';
+import {getThemeValue} from './utilities';
+import {
+  BaseTheme,
+  Dimensions,
+  ResponsiveBaseTheme,
+  RestyleFunction,
+  RNStyleProperty,
+  StyleTransformFunction,
+} from './types';
+import {getResponsiveValue} from './utilities/getResponsiveValue';
+
+const getMemoizedMapHashKey = (
+  dimensions: Dimensions | null,
+  themeKey?: string,
+  property?: string,
+  value?: string | number | boolean,
+) => {
+  return `${dimensions?.height}x${dimensions?.width}-${themeKey}-${property}-${value}`;
```

Comment:
when debugging the memoized map keys I found out that sometimes `themeKey` `property` or `value` could be `undefined`. I would an extra check as this could lead to false hits for the cache.

Reply:
I don't think that would be a false hit or? If a `value` is undefined for given key, property, and dimensions, it should be undefined at all times.

Reply:
I was seeing keys like `300-300-undefined-padding-undefined` (not remember the specific case). I am worried this could lead to store the values of different rules and lead to inconsistencies. I would return `undefined` if any of the keys are undefined and not use the cache with this specific cases. 

We can also investigate specific cases on why is this happening, but I would keep this investigation for later to not block this PR.

Reply:
Fair, addressed